osu.Game/Rulesets/Edit/Checks/Components/IssueType.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace osu.Game.Rulesets.Edit.Checks.Components
+{
+    /// <summary>
+    /// The type, or severity, of an issue. This decides its priority.
+    /// </summary>
+    public enum IssueType
+    {
+        /// <summary> A must-fix in the vast majority of cases. </summary>
+        Problem = 3,
+
+        /// <summary> A possible mistake. Often requires critical thinking.  </summary>
+        Warning = 2,
+
+        // TODO: Try/catch all checks run and return error templates if exceptions occur.
+        /// <summary> An error occurred and a complete check could not be made. </summary>
+        Error = 1,
+
+        // TODO: Negligible issues should be hidden by default.
+        /// <summary> A possible mistake so minor/unlikely that it can often be safely ignored. </summary>
+        Negligible = 0,
```

Comment:
Not sure about the reverse ordering here. Is that done for a reason? Probably best to order ascending to follow standard (and apply sorting at some other level if it's required)